### PR TITLE
Feature store

### PR DESCRIPTION
**Pull request by amanbhatt26**

### CODE REVIEW

File: src/Components/Answer/Answer.tsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { Link } from "react-router-dom";
 
 export type AnswerProps = {
   votes: number;
-  question: string;
+  question?: string | null | undefined;
```
Comment (Suparv) [inline src/Components/Answer/Answer.tsx:7]: why is question optional

File: src/Components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -1,13 +1,28 @@
+import { useDispatch, useSelector } from "react-redux";
 import { Link } from "react-router-dom";
 import { HomeIconOutline, HomeIconSolid, QuestionIconOutline } from "../Icons";
+import { changeScreen } from "../../Features/Navigation/NavSlice";
+import { link } from "fs";
 
 export const Navbar = () => {
+  const { tab } = useSelector((state: any) => state.navigation);
```
Comment (Suparv) [inline src/Components/Navbar/Navbar.tsx:8]: Avoid any

File: src/Components/Question-Screen/QuestionScreen.tsx [hunk 3]
```diff
@@ -1,67 +1,40 @@
-import { Question } from "../Question/Question";
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { changeScreen } from "../../Features/Navigation/NavSlice";
+import { Question, QuestionProps } from "../Question/Question";
 
 export const QuestionScreen = () => {
+  const { questions } = useSelector((state: any) => state.questionScreen);
```
Comment (Suparv) [inline src/Components/Question-Screen/QuestionScreen.tsx:7]: avoid any

File: src/Components/Question-detail-Screen/QuestionDetail.tsx [hunk 4]
```diff
@@ -1,57 +1,57 @@
-import { WithoutQuestion } from "../Answer";
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { changeScreen } from "../../Features/Navigation/NavSlice";
+import { Answer, AnswerProps } from "../Answer";
 import { Question } from "../Question/Question";
 
 export const QuestionDetail = () => {
+  const { question, answers } = useSelector(
+    (state: any) => state.questionDetail
```
Comment (Suparv) [inline src/Components/Question-detail-Screen/QuestionDetail.tsx:9]: avoid any

File: src/Features/User-Screen/UserSlice.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { UserProps } from "../../Components/User-Screen";
+
+const initialState: UserProps = {
+  username: "Aman Bhatt",
+  profession: "Web Developer",
+  joinedAt: "2 days ago",
```
Comment (Suparv) [inline src/Features/User-Screen/UserSlice.tsx:7]: backend will return a dateTime epoch, write a function to convert epoch to string in utils ?

**Review by Suparv — CHANGES_REQUESTED** [review]

File: src/Components/Answer/Answer.tsx (hunk 1 above)
Comment (amanbhatt26) [reply src/Components/Answer/Answer.tsx:7]: the answer component has two usages , one is with question and one is without

**Pull request closed without merging**